### DISCUSSION TITLE
Git add safe.directory fails on UNC path

### DISCUSSION THREAD

**Discussion by maphew** [gitextensions/gitextensions#10907] [General]
### Environment

- GitExtensions version: 4.0.2.16100
- GIT version: 2.39.1.windows.1
- OS version: Windows 10.0.19045 Build 19045
- .NET version: 

### Issue description

Responding to the prompt "yes" add this folder to gobal safe dirs appears to do the right thing but fails. After several attempts opening the global `.gitconfig` file shows repeated entries for the same folder.

The upstream git issue seems to be that single or double quote are needed _differently_ depending on whether the `git -glocal -add` command is being run from Git Bash, Powershel, or CMD.

See https://github.com/git-for-windows/git/issues/3856

### Steps to reproduce

Open a repository on \\server\share\folder that is not owned by current user.
Answer Yes to prompt to add to safe exceptions

### Diagnostics

Manually editing .gitconfig and wrapping the dir in both single and double quotes with different entries seems to workaround the issue. Ex:

```
[safe]
	directory = "%(prefix)///Server/Maps/Project558/scripts"
	directory = '%(prefix)///Server/Maps/Project558/scripts'
```

**Comment by mstv** [comment]
Thank you for sharing the workaround and the git issue, @maphew.
Though I hesitate to implement it in GE. Let's wait for a more consistent handling by Git.

**Comment by maphew** [comment]
It looks like their fix is just going to be a usage advisory notice according to https://github.com/git-for-windows/git/issues/3856#issuecomment-1129278897, and the issue hasn't been changed for almost a year. What shell does GE call for it's commands? Does it specify bash/pwsh/cmd or is that up to OS settings?

**Comment by mstv** [comment]
GE starts git.exe as a standard Windows process without a shell.
Though since it is likely that git commands are run manually from bash, too, we should add both variants.
But GE uses the command suggested by git. This is done in the method `BugReportInvoker.ReportDubiousOwnership()`.
Creating and executing a second adapted command will need some handling and tests.
A PReq will be welcome...
